File: raehyeon/BOJ/Silver/4/큐_2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        Queue<Integer> q = new LinkedList<>();

        int n = Integer.parseInt(br.readLine());
        int end = 0; // 큐의 마지막 값을 저장하기 위한 변수

        for(int i = 0; i < n; i++){
            st = new StringTokenizer(br.readLine());
            String func = st.nextToken();

            switch(func) {
                case"push":
                    int num = Integer.parseInt(st.nextToken());
                    q.offer(num);
                    end = num;
                    break;

                case "pop":
                    if(q.isEmpty()) {
                        sb.append(-1).append("\n");;
                    } else {
                        sb.append(q.poll()).append("\n");
                    }
                    break;

                case "size":
                    sb.append(q.size()).append("\n");
                    break;

                case "empty":
                    if(q.isEmpty()) {
                        sb.append(1).append("\n");
                    } else {
                        sb.append(0).append("\n");
                    }
                    break;

                case "front":
                    if(q.isEmpty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(q.peek()).append("\n");
                    }
                    break;

                case "back":
                    if(q.isEmpty()) {
                        sb.append(-1).append("\n");
                    } else {
                        sb.append(end).append("\n");
                    }
                    break;
            }
        }

        System.out.println(sb);
        br.close();
    }
}

File: syJoh/BOJ/큐2.py
import sys
from collections import deque

input = sys.stdin.readline
n = int(input())
q = deque()
num = 0
answer = []
for _ in range(n):
    word = list(input().split())
    if word[0] == 'push':
        num = word[1]
        q.append(num)
    elif word[0] == 'pop':
        if len(q) == 0:
            answer.append(-1)
        else:
            answer.append(q.popleft())
    elif word[0] == 'size':
        answer.append(len(q))
    elif word[0] == 'empty':
        if len(q)==0:
            answer.append(1)
        else:
            answer.append(0)
    elif word[0] == 'front':
        if len(q)==0:
            answer.append(-1)
        else:
            answer.append(q[0])
    elif word[0] == 'back':
        if len(q)==0:
            answer.append(-1)
        else:
            answer.append(num)

for i in answer:
    print(i)

"""
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;
        int n = Integer.parseInt(br.readLine());
        int tmp=0;
        Queue<Integer> que = new LinkedList<>();

        while(n>0){
            st = new StringTokenizer(br.readLine());
            String order = st.nextToken();
            if(order.equals("push")){
                tmp = Integer.parseInt(st.nextToken());
                que.offer(tmp);
            }else if(order.equals("pop")){
                if(que.isEmpty()){
                    bw.write(String.valueOf(-1));
                }else{
                    bw.write(String.valueOf(que.poll()));
                }
                bw.newLine();
            }else if(order.equals("size")){
                bw.write(String.valueOf(que.size()));
                bw.newLine();
            }else if(order.equals("empty")){
                if(que.isEmpty()){
                    bw.write(String.valueOf(1));
                }else{
                    bw.write(String.valueOf(0));
                }
                bw.newLine();
            }else if(order.equals("front")){
                if(que.isEmpty()){
                    bw.write(String.valueOf(-1));
                }else{
                    bw.write(String.valueOf(que.peek()));
                }
                bw.newLine();
            }else if(order.equals("back")){
                if(que.isEmpty()){
                    bw.write(String.valueOf(-1));
                }else{
                    bw.write(String.valueOf(tmp));
                }
                bw.newLine();
            }
            n--;
        }
        bw.flush();
        br.close();
        bw.close();
    }
}
"""
